homeassistant/components/system_bridge/coordinator.py

```diff
@@ -49,97 +72,109 @@ def update_listeners(self) -> None:
         for update_callback in self._listeners:
             update_callback()
 
-    async def async_handle_event(self, event: Event):
-        """Handle System Bridge events from the WebSocket."""
-        # No need to update anything, as everything is updated in the caller
-        self.logger.debug(
-            "New event from %s (%s): %s", self.title, self.host, event.name
-        )
-        self.async_set_updated_data(self.bridge)
+    async def async_handle_data(
+        self,
+        module: str,
+        data: dict,
+        module_handler: Callable,
+    ) -> None:
+        """Handle data from the WebSocket client."""
+        if module in MODULES:
+            self.logger.debug("Set new data for: %s", module)
+            setattr(self.systembridge_data, module, module_handler(**data))
```

Comment:
The library user shouldn't need to create the model. That's the responsibility of the library.